docs/DesignDocs/CustomAttestation.md

```diff
@@ -0,0 +1,420 @@
+___
+
+Custom Attestation Data Formats for Open Enclave
+=====
+
+This design document proposes a new attestation framework and set of APIs that
+enable developers to use custom formats for their attestation evidence and
+collateral.
+
+Motivation
+----------
+
+Currently, Open Enclave provides several APIs that developers can use for
+attestation. The two key functions are `oe_get_report`, which produces an opaque
+blob that is signed by the enclave, and `oe_verify_report`, which can be used to
+verify the generated report. The original purpose of those two APIs were to
+provide a simple, cross-platform way to produce and verify attestation data.
+
+However, for some developers, these APIs are too inflexible for their
+attestation requirements. For example, one might want to extend Open Enclave's
+current attestation structures with extra information, such as geolocation
+or a timestamp. Another user might want their enclaves to generate attestation
+data that is in a compatible format with their existing authentication
+infrastructure, such as a JSON Web Token or a X509 certificate. Overall,
+there has been interest in enchancing Open Enclave's APIs to support custom
+attestation formats to enable these scenarios.
+
+Specification
```

Comment:
Maybe explain a little bit about what claims are and how can they be used in this section?

Reply:
I'll add a terminology section.